### ISSUE TITLE
🚀 Feature: Keep the original file extension for files in the uploads/shares/<sharename> folder

### ISSUE THREAD

**Issue by b-a0** [stonith404/pingvin-share#342]
### 🔖 Feature description

Currently uploaded files are structured as follows when a share is created:

```
data
|-- uploads
    |-- shares
        |-- share1
            |-- long-hash1
            |-- long-hash2
            |-- long-hash3
            |-- long-hash4
            |-- archive.zip
```

The files under `share1` have no file extension. The currently described feature would keep the long hash filename, but preserve the original file extension of the uploaded files, such that the filetree would look as follows:

```
data
|-- uploads
    |-- shares
        |-- share1
            |-- long-hash1.jpeg
            |-- long-hash2.JPG
            |-- long-hash3.PNG
            |-- long-hash4.JPG
            |-- archive.zip
```

### 🎤 Pitch

Using pingvin share I would like to create reverse shares to request photo uploads from a group of people (ideally in combination with the features requested in #341 and #155). I would then like to import those photos into [photoprism](https://www.photoprism.app/) automatically. In photoprism I can [set a folder to watch](https://docs.photoprism.app/user-guide/library/import/) (in this case that would be the pingvin `/data/upload/shares` folder), but it only processes files that have a valid extension.

In general it would be nice to have the uploaded files in the `share` folder "ready to go", such that other tools can pick them up and process them (without having to guess file extensions).

**Comment by plague-doctor** [comment]
It would be great if this can activate per share, and not as a default for all shares.

**Comment by b-a0** [comment]
> It would be great if this can activate per share, and not as a default for all shares.

Out of curiosity: what is the downside of enabling this for all shares?

**Comment by plague-doctor** [comment]
Privacy. I am happy with the current, scrambled file names. However, you need to have the extensions in readable form. 
I am happy with all shares with decrypted extensions as long as I can switch feature this off ;-)

**Comment by CrazyWolf13** [comment]
Yeah I'd also love to see this implemented, or even to fully disable FIlename Hashing at all, as I only use locally and behind auth server, that would make many processes much easier

**Comment by adamzvolanek** [comment]
> Yeah I'd also love to see this implemented, or even to fully disable FIlename Hashing at all, as I only use locally and behind auth server, that would make many processes much easier

I wonder if there's already a feature request for this. It would be nice to not have the files display by their hash. Feature that can be controlled by a feature flag/environment variable.
